### ISSUE TITLE
NetMQ SUB cannot receive from JeroMQ PUB

### ISSUE THREAD

**Issue by jasells** [zeromq/jeromq#750]
I believe I have found the inverse of this issue (negotiating of the ZMQ protocol v 3 down to v2) on a "server" (my test case is JeroMQ PUB):

https://github.com/zeromq/jeromq/pull/461

I am trying to connect  NetMQ SUB to a JeroMQ PUB, and the NetMQ client never sees any topics from TryReceive, nor does ReceiveReady ever fire.

I re-ran the test scenario with the ZeroMQ C# wrapper package, and it works with the same 
JeroMQ PUB socket. (Presumably because they both can use protocol v3 and don't need to negotiate down?)

Wireshark shows that the JeroMQ PUB is sending messages even though the NetMQ SUB never tells the application layer it has data....

A NetMQ SUB is able to connect to NetMQ PUB.
JeroMQ is v 5.1

*************Update:
We tried JeroMQ versions 4.1, 4.3, 5.0, 5.1
All behave the same way.

I think the JeroMQ issue linked to above was only solved for the JeroMQ client side, and now we have exposed the same or similar issue on the JeroMQ server side?

I say this is on JeroMQ side because NetMQ SUB *does* work with ZeroMQ C# PUB (which is the reference c implementation wrapped in C# for API only).

Also, see here as ref: 
https://github.com/zeromq/netmq/issues/693

##########update 2
JeroMQ SUB. _works_ with NetMQ PUB...…. Seems to only be the reverse has problem

**Comment by trevorbernard** [comment]
I would suggest trying to use a version of `libzmq` that doesn't supports v3 ZMTP wire protocol. I believe it's in the 3.2.x releases. https://github.com/zeromq/zeromq3-x/releases

**Comment by jasells** [comment]
Use libzmq instead of jeromq?  Latest libzmq works as expected, not supporting v3 at all is not the right answer, though could be a band aid.

The code base that needs to be server is all Java.... so that's why we picked jero in first place.

It _seems_ that v3 of ZMPT wire protocol support is _supposed_ to include backward compatibility....

**Comment by trevorbernard** [comment]
It's for debugging purposes to help verify your findings that it is indeed an upgrade/downgrade issue. Libzmq is the canonical implementation of the ZMTP protocol and NetMQ and JeroMQ are ports. It's possible the bug exists in all threw projects

**Comment by jasells** [comment]
@trevorbernard 


> It's for debugging purposes to help verify your findings that it is indeed an upgrade/downgrade issue. Libzmq is the canonical implementation of the ZMTP protocol and NetMQ and JeroMQ are ports. It's possible the bug exists in all threw projects


Understood. I did that before posting.   clrzmq is just a C# api warapper around libzmq.  Sorry that was clear, but that's what I meant by 
>ZeroMq C#

*Both* jeroMQ. And NetMQ work with clrzmq/libzmq.

**Comment by daveyarwood** [comment]
https://github.com/zeromq/jeromq/issues/631 would help us set up a test bed for issues like this.

**Comment by somdoron** [comment]
NetMQ now supports ZMTPv3 (pre-released versions), so you might want to check again

**Comment by GABowers** [comment]
I don't know the specifics of how "clients" and "servers" work for Request/Response, but I just added the latest org.zeromq.jeromq version from maven (https://mvnrepository.com/artifact/org.zeromq/jeromq) to a Java project and set up a ZMQ.REQ socket. I also set up a ResponseSocket in C# using the latest version of NetMQ from nuget (https://www.nuget.org/packages/NetMQ/).

From Java I sent a multipart message, the C# received it and sent a response, and the Java code received the response.
